File: backend/entity/Announcement.go
//ประกาศ
package entity

import (
	"gorm.io/gorm"
	"time"
)
type StatusAnnouncement string
const (
		Publish StatusAnnouncement = "เผยแพร่แล้ว"
		Draft	StatusAnnouncement = "ฉบับร่าง"
)
type Announcement struct {
	gorm.Model
	Title  			string 				`json:"title"`
	Create_Date 	time.Time			`json:"create_date"`
	End_Date 		time.Time			`json:"end_date"`
	TimeCreate		string				`json:"time_create"`
	Content 		string				`json:"content"`
	Category        string				`json:"category"`
	Status 			StatusAnnouncement	`gorm:"default:'ฉบับร่าง'" json:"status"`
	
	TargetGroupID uint
	Target_Group *Target_Group `gorm:"foreignKey:TargetGroupID" json:"target_group"`

	AdminID uint
	Admin_User *Admin_User `gorm:"foreignKey:AdminID"`

	TermID uint
	Term *Term `gorm:"foreignKey:TermID"`

	EnrollmentID uint
	Enrollment *Enrollment `gorm:"foreignKey:EnrollmentID" json:"enrollment"`

	EnrollmentsID uint
	Enrollments *Enrollments `gorm:"foreignKey:EnrollmentsID" json:"enrollments"`
	Users []Users `gorm:"many2many:notifications" json:"users"` // many to many กับ Users ผ่านตาราง Notifications
}
func ParseTime(timeStr string) time.Time {
	layout := "15:04"
	t, err := time.Parse(layout, timeStr)
	// return time.Now()
	// parsedTime, err := time.Parse(time.RFC3339, timeStr)
	if err != nil {
		// ถ้าแปลงไม่ได้ ให้ใช้เวลาปัจจุบัน
		return time.Now()
	}
	return t
}
func ParseDate(dateStr string) time.Time {
	layout := "2006-01-02"
	parsedDate, err := time.Parse(layout, dateStr)
	if err != nil {
		return time.Time{}
	}
	return parsedDate
}

File: frontend/src/interfaces/announcement.ts
import type { TargetGroupInterface } from "./targetgroup";
export interface AnnouncementInterface {
    ID?:                number
    title?:             string;
    content?:           string;
    category?:          string;
    status?:            string;
    create_date?:       string;
    end_date?:          string;
    target_group_id?:   TargetGroupInterface;
    group_name?:        string;
    user_id?:           number;
    admin_id?:          number;
    term_id?:           number;
    enrollment_id?:     number;
    TargetGroupID?:     number;
    time_create?:       string;
    target_group?:      TargetGroupInterface;
    
}


// Title         string `json:"title" binding:"required"`
// 	Content       string `json:"content" binding:"required"`
// 	Category      string `json:"category" binding:"required"`
// 	Status        string `json:"status"`         // เช่น "ฉบับร่าง" หรือ "เผยแพร่แล้ว"
// 	CreateDate    string `json:"create_date"`    // ควรเก็บเป็น time.Time ใน entity
// 	EndDate       string `json:"end_date"`       // ควรเก็บเป็น time.Time
// 	TargetGroupID uint   `json:"target_group_id" `
// 	UserID        uint   `json:"user_id"` // ผู้รับแจ้งเตือน
// 	AdminID       uint   `json:"admin_id"` // ผู้สร้างประกาศ
// 	TermID        uint   `json:"term_id"`
// 	EnrollmentID  uint   `json:"enrollment_id"`